<launch>
  <arg name="robot" default="/"/>
  <arg name="base" value="kobuki"/>
  <param name="use_sim_time" value="true"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="use_sim_time" value="true"/>
    <arg name="debug" value="false"/>
    <arg name="world_name" value="$(find neato_simulator)/worlds/empty.world"/>
  </include>

  <group ns="robot1">
    <arg name="urdf_file" default="$(find xacro)/xacro.py '$(find neato_simulator)/model/neato.urdf.xacro'" />
    <param name="robot_description" command="$(arg urdf_file)" />

    <!-- Gazebo model spawner -->
    <node name="spawn_neato_model" pkg="gazebo_ros" type="spawn_model"
        args="-y 0 -unpause -urdf -param robot_description -model robot1/mobile_base"/>

    <!-- laser filter -->
    <node name="laser_filter" pkg="neato_simulator" type="laser_filter.py"/>

    <!-- topic relayer -->
    <node name="relay_velocity_cmds" pkg="topic_tools" type="relay" args="cmd_vel mobile_base/commands/velocity"/>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
      <param name="publish_frequency" type="double" value="30.0" />
    </node>
  </group>
  <group ns="robot2">
    <arg name="urdf_file" default="$(find xacro)/xacro.py '$(find neato_simulator)/model/neato.urdf.xacro'" />
    <param name="robot_description" command="$(arg urdf_file)" />

    <!-- Gazebo model spawner -->
    <node name="spawn_neato_model" pkg="gazebo_ros" type="spawn_model"
        args="-y 1 -unpause -urdf -param robot_description -model robot2/mobile_base"/>

    <!-- laser filter -->
    <node name="laser_filter" pkg="neato_simulator" type="laser_filter.py"/>

    <!-- topic relayer -->
    <node name="relay_velocity_cmds" pkg="topic_tools" type="relay" args="cmd_vel mobile_base/commands/velocity"/>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
      <param name="publish_frequency" type="double" value="30.0" />
    </node>
  </group>
</launch>
